internal/stream/event_stream.go

```diff
@@ -0,0 +1,94 @@
+package stream
+
+import (
+	"context"
+	"errors"
+	"reflect"
+
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+type EventStream struct {
+	writer EventStreamWriter
+	log    logrus.FieldLogger
+}
+
+const (
+	TypeInfraEnv = "InfraEnv"
+	TypeHost     = "HostState"
+	TypeCluster  = "ClusterState"
+	TypeEvent    = "Event"
+)
+
+type Resource interface{}
```

Comment:
Is there not an existing interface we can use? Like a gorm base model or some swagger thing?